programs/ssl/ssl_client2.c

```diff
@@ -438,6 +438,8 @@ static int my_send( void *ctx, const unsigned char *buf, size_t len )
 }
 
 #if defined(MBEDTLS_X509_CRT_PARSE_C)
+static unsigned char peer_crt_info[1024] = { 0 };
```

Comment:
Major: It seems to me that this buffer needs to be zeroized not only here, but every time the client is preparing for a new handshake (for example when the reconnect option is used), however I can't see any code doing that later in the file.

Reply:
In an ordinary run, I don't think it would need zeroization at all because it's always copied from `buf` in `my_verify()` - isn't it? Yet I found is safer to initialize it in any case to have it represent a well-formed null-terminated string at any time.

Reply:
What if a the first handshake uses a certificate but the second (in case of reconnect) doesn't? In that case `my_verify()` won't be called in the second handshake, so after the second handshake `peer_crt_info` will contain the cert info from the first handshake, while it should arguably be the empty string at this point. Am I missing something?

Reply:
@mpg Yes, good point! I thought about whether or not `peer_crt_info` would always be a well-formed string, but not whether it could become outdated. I'll change it.

Reply:
Done.